# mutational_scanning and QCC
Analysis of variants generated by mutational scanning for a `Methods in MolecularBiology` book chapter ([Directed Evolution of Proteins Based on Mutational Scanning](https://link.springer.com/protocol/10.1007/978-1-4939-7366-8_6)).

## mutational scanning
`deep_mut_scanning.py` is a python script to generate a list of partially overlapping QuikChange primers aimed at deep mutational scanning experiments.     
It iterates across a DNA sequence, codon by codon and generates a primer pair that employs the partially overlapping primer pair strategy [ref](http://nar.oxfordjournals.org/content/43/2/e12/F3.expansion.html). Namely, the primers in a pair will have an user-defined overlap length (e.g. 22 bp) centred around the codon to mutate and will have a 3â€™ overhand long enough to allow the region beyond the mutagenized codon to anneal with the template above a given melting temperature, while taking into account terminal GC clamp. It is also available as a web app ([deepscan.matteoferla.com/](http://deepscan.matteoferla.com/)).

## P450 Analysis
In the main text of the chapter an example dataset is analysed.
One of the figures requires [ternarty plot from MEx](http://uk.mathworks.com/matlabcentral/fileexchange/7210-ternary-plots).

## QQC
Two Matlab files are present to illustrate how one does a QQC analysis.
One is a fully automated script, while the other requires the user to choose the file.
### Annoyance
As Matlab cannot open ab1 files, they need to be converted. For this Staden tools library is used to convert ab1 to scf.    
* Download from [sourceforge](https://sourceforge.net/projects/staden/files/)
* unzip
* On a mac OS X open terminal
* navigate to the folder, probably this: `$ cd ~/Downloads/io_lib-1.14.6` (don't type the dollar, it is there already')
* read the instructuctions and copyright notice `$ cat README`
* do as they say `$ ./configure` then `$ make` then `$ make install`

### scfread
Matlab can read the non-propriety scf thanks to the function `scfread`. Which is documented [here](https://uk.mathworks.com/help/bioinfo/ref/scfread.html).    

Confusingly, flow citometry files are `fsc` files, which have nothing to do with this, and to further the confusion the function from [Exchange `fca_readfsc`](http://uk.mathworks.com/matlabcentral/fileexchange/9608-fcs-data-reader) to read them has the words arranged backwards. So if fsc files sound familiar it because of that probably.    
As explained in the manual, two files are returned with the call:   
`[Sample, Probability] = scfread(fullfile(pathname,'temp.scf'));`    
`Sample` contains all the fluorescent channels data, while `Probability` has the elaborated versions.
### Analysis
To see how a QQC analysis is done see the `.m` files `QQC_manual.m` and `QQC_fully_automated.m`.

## Licence
In brief, AFAIK the copyleft GNU licence means you can use, change and redistribute the code if you cite it (_i.e._ normal academic practice).


